robot_sort/robot_sort.py

```diff
@@ -97,7 +97,29 @@ def sort(self):
         Sort the robot's list.
         """
         # Fill this out
-        pass
+        self.set_light_on()
+        while self.light_is_on() is True:
+            self.set_light_off()
+
+            while self.can_move_right() is True:
+                self.swap_item()
+                self.move_right()
+
+                if self.compare_item() == 1:
+                    self.swap_item()
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+                    self.set_light_on()
+
+                else:
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+
+            if self.light_is_on():
+                while self.can_move_left():
+                    self.move_left()
```

Comment:
good job with this. an alternative would be to implement a selection sort which could be done like this:

self.swap_item()
self.set_light_on()

while self.light_is_on():
(move right and find the smallest item that has not been sorted)
____while self.move_right():
________if self.compare_item() > 0:
____________self.swap_item()
( if the robot is at the end of the list and all items are sorted, we're done)
________if self.compare_item() is None and self.can_move_right() is False:
(place the current item down in the list and turn light off to indicate the list is sorted)
____________self.swap_item()
____________self.set_light_off()
____________break
________else:
( find the spot in the list that is holding None)
( place the item in that position)
(move right and swap)
____________while self.move_left():
________________if self.compare_item() is None:
________________self.swap_item()
________________self.move_right()
________________self.swap_item()
________________break